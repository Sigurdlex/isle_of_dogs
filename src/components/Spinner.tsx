import React from 'react';
import styled from 'styled-components';

interface SpinnerProps {
  color: string;
}

export const Spinner = ({ color }: SpinnerProps) => <Container color={color} />;

const Container = styled.div`
  color: #ffffff;
  font-size: 11px;
  margin: 55px auto;
  top: 50%;
  left: 50%;
  width: 10em;
  height: 10em;
  box-shadow: inset 0 0 0 1em;
  transform: translate(-50%, -50%);

  &,
  &::before,
  &::after {
    border-radius: 50%;
    position: absolute;
  }

  &::before,
  &::after {
    content: '';
    background: ${({ color }: SpinnerProps) => color};
    width: 5.2em;
    height: 10.2em;
  }

  &::before {
    border-radius: 10.2em 0 0 10.2em;
    top: -0.1em;
    left: -0.1em;
    transform-origin: 5.2em 5.1em;
    animation: load2 2s infinite ease 1.5s;
  }

  &::after {
    border-radius: 0 10.2em 10.2em 0;
    top: -0.1em;
    left: 5.1em;
    transform-origin: 0 5.1em;
    animation: load2 2s infinite ease;
  }

  @keyframes load2 {
    0% {
      transform: rotate(0deg);
    }
    100% {
      transform: rotate(360deg);
    }
  }
`;
