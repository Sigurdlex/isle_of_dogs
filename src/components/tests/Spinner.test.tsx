import React from 'react';
import { render, cleanup } from '@testing-library/react';
import 'jest-styled-components';

import { Spinner } from '../Spinner';

afterEach(cleanup);

test('<Spinner />', () => {
  const { container } = render(<Spinner color="white" />);
  const element: HTMLElement = container.firstChild as HTMLElement;
  expect(element).toMatchSnapshot();
});
