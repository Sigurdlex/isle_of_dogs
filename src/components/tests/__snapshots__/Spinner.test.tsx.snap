// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Spinner /> 1`] = `
.c0 {
  color: #ffffff;
  font-size: 11px;
  margin: 55px auto;
  top: 50%;
  left: 50%;
  width: 10em;
  height: 10em;
  box-shadow: inset 0 0 0 1em;
  -webkit-transform: translate(-50%,-50%);
  -ms-transform: translate(-50%,-50%);
  transform: translate(-50%,-50%);
}

.c0,
.c0::before,
.c0::after {
  border-radius: 50%;
  position: absolute;
}

.c0::before,
.c0::after {
  content: '';
  background: white;
  width: 5.2em;
  height: 10.2em;
}

.c0::before {
  border-radius: 10.2em 0 0 10.2em;
  top: -0.1em;
  left: -0.1em;
  -webkit-transform-origin: 5.2em 5.1em;
  -ms-transform-origin: 5.2em 5.1em;
  transform-origin: 5.2em 5.1em;
  -webkit-animation: load2 2s infinite ease 1.5s;
  animation: load2 2s infinite ease 1.5s;
}

.c0::after {
  border-radius: 0 10.2em 10.2em 0;
  top: -0.1em;
  left: 5.1em;
  -webkit-transform-origin: 0 5.1em;
  -ms-transform-origin: 0 5.1em;
  transform-origin: 0 5.1em;
  -webkit-animation: load2 2s infinite ease;
  animation: load2 2s infinite ease;
}

<div
  class="c0"
  color="white"
/>
`;
